```{r, error=TRUE, results='hide', message=FALSE}
library(SummarizedExperiment)
library(org.Mm.eg.db)
library(org.Hs.eg.db)
library(DEFormats)
library(tximportData)
library(tximport)
library(rhdf5)
library(SummarizedExperiment)
library("tximeta")
#library("SummarizedExperiment", warn.conflicts = FALSE)
library("edgeR")
library("limma")
library(tximport)
library(GenomicFeatures)
library(BiocParallel)
library(org.Hs.eg.db)
library(Homo.sapiens)
library(Homo.sapiens)
library("sva")
library(DESeq2)
library("tximeta")
library("edgeR")
library(Rsubread)
library(BiocParallel)
library(tximportData)
library(tximport)
library(rhdf5)
library(SummarizedExperiment)
library("tximeta")
library("SummarizedExperiment", warn.conflicts = FALSE)
library("edgeR")
library("limma")
library(tximport)
library(GenomicFeatures)
library(BiocParallel)
library(org.Hs.eg.db)
library(Homo.sapiens)
library("sva")
library(RColorBrewer)
library(Glimma)
library(gplots)

```




```{r, error=TRUE}

extract_geneid_parallel <- function(geneid){
  
  return(strsplit(geneid,".", fixed = TRUE)[[1]][1])
}

coldata <- read.csv("~/Documents/salmon_v34/sample_table_cells2.csv", row.names=1, stringsAsFactors=FALSE )
coldata$names <- coldata$Run
#files <- file.path("~/Documents/Analiza_RNA_seq/Kallisto/kallisto_v33", coldata$names, "abundance.h5" )
#file.exists(files)
```


```{r, error=TRUE}
x<- read.delim("~/Documents/Analiza_RNA_seq/analiza_RNA/counts_cells.txt")

names(x)
## Take a look at the featurecounts stats
x$stat
## Take a look at the dimensions to see the number of genes
dim(x$counts)
head(x$counts)
head(x$annotation)
```

```{r, error=TRUE}
counts_subread <- x[,-c(1,2)]
rownames(counts_subread) <- x[,1]
Group <- factor(paste(coldata$Line,coldata$Condition,sep="."))

y <- DGEList(counts_subread, group=Group, genes=x[,1,drop=FALSE])
keep <- filterByExpr(y, group=Group)#etap filtracji
y_filter <- y[keep, , keep.lib.sizes=FALSE]

subread.g.filtered  <- y[keep, keep.lib.sizes=FALSE]
group <- factor(coldata$name)
samples <- coldata$name
```


# ```{r, error=TRUE}
# Group <- factor(paste(coldata$Line,coldata$Condition,sep="."))
# 
# y <- DGEList(fc$counts, group=Group)
# keep <- filterByExpr(y, group=Group)#etap filtracji
# y_filter <- y[keep, , keep.lib.sizes=FALSE]
# 
# subread.g.filtered2  <- y[keep, keep.lib.sizes=FALSE]
# group <- factor(coldata$name)
# samples <- coldata$name
# 
# ```


Fragment uzyskania genes do gene ontology
```{r, error=TRUE, results='hide', message=FALSE}
yz <- DGEList(counts=subread.g.filtered, group = group)
x <- yz
dim(x)
geneid <- rownames(x)
#geneids <- bplapply(geneid, extract_geneid_parallel)
geneids <- lapply(geneid,  sub, pattern = "\\.\\d+$", replacement = "")

geneids <- unlist(geneids, use.names=FALSE)
geneids
keytypes(Homo.sapiens)
genes <- select( Homo.sapiens, keys=geneids, columns=c("SYMBOL", "TXCHROM", "GENENAME", "ENTREZID"), 
                 keytype="ENSEMBL")

annot <- select(Homo.sapiens, keys=geneids, "ENTREZID", keytype="ENSEMBL")
annot <- annot[!duplicated(annot$ENSEMBL),]

x   
head(genes)
options(max.print=100000)
genes
genes <- genes[!duplicated(genes$ENSEMBL),]
x$genes <- genes
x 



gene_entrezid <- mapIds(Homo.sapiens, keys=geneids,
                   keytype="ENSEMBL", column="ENTREZID")

gene_symbol <- mapIds(Homo.sapiens, keys=geneids,
                   keytype="ENSEMBL", column="SYMBOL")

gene_names <-  mapIds(Homo.sapiens, keys=geneids,
                   keytype="ENSEMBL", column="GENENAME")

gene_txid <- mapIds(Homo.sapiens, keys=geneids,
                   keytype="ENSEMBL", column="ENSEMBLTRANS")

genes_df <- data.frame(ENSEMBL=rownames(subread.g.filtered), ENTREZID=gene_entrezid, SYMBOL=gene_symbol, GENENAME=gene_names, ENSEMBLTRANS=gene_txid)

genes_df


```


```{r, error=TRUE}
dge <- DGEList(subread.g.filtered, group=Group, genes=genes_df)

dge <- calcNormFactors(dge)
```



Exploring differences between libraries
```{r, error=TRUE}

pch <- c(0,1,2,15,16,17, 18, 19 , 20)
colors <- rep(c("darkgreen", "red", "blue", "green", "orange"), 2)
plotMDS(y, col=colors[Group], pch=pch[Group])
legend("bottom", legend=levels(Group), pch=pch, col=colors, ncol=2)

plotMDS(y, col=colors[Group], pch=pch[Group])

```
```{r, error=TRUE}

library(Glimma)
glMDSPlot(y, labels=paste(Group), 
          groups=Group, launch=TRUE)
```



```{r, error=TRUE}
for (i in 1:ncol(y)){
  plotMD(y, column=i)
  abline(h=0, col="red", lty=2, lwd=2)
}

```



Modele do SVA
```{r, error=TRUE, results='hide', message=FALSE}
mod1 <- model.matrix(~0+Group)
mod1
mod0 <- model.matrix(~1, coldata)
```



Tutaj SVA tak jak w skrypcie który dostaliśmy, sv na końcu wychodzi 1, metoda leek, 
```{r, error=TRUE, results='hide', message=FALSE}
rownames(mod1) <- colnames(subread.g.filtered)
v <- voom(dge,mod1,plot=FALSE)
to_sva <- 2^v$E # do sva  dajemy znormalizowane, 
#nie zlogarytmowane dane! ( voom robi log2- dlatego wracamy do oryginalnej skali)
nsv<-num.sv(as.matrix(subread.g.filtered),mod1,method="leek") 

#Tutaj jest pztla, ze wzgledu na to, ze svaseq zazwyczaj wyrzuca blad jezeli 
#dasz mu tyle faktorów, ile num.sv pokaze na poczatku-  lecimy az savseq zadziala poprawnie
an.error.occured <- TRUE
while(an.error.occured==TRUE){
  svseq <- try(svaseq(to_sva,mod1,mod0,const=0,n.sv=nsv),TRUE)
  if(isTRUE(class(svseq)=="try-error")) { 
    nsv <-nsv-1
    next
  } 
  else { an.error.occured <- FALSE}
}
newV=NULL
fsvaobj<-fsva(v$E,mod1,svseq,newdat=NULL)
mmod1 <- cbind(mod1, svseq$sv)#mmod1 to design
#colnames(mmod1) <- c(levels(factor(samples)), paste("X",1:nsv,sep=""))
v2 <- voom(dge,mmod1,plot=FALSE)



# Ta pętla jest z kolei dlatego, że często lmfit też wyrzyuca błąd przy zbyt dużej liczbie faktorów,
# nawet, jeśli svaseq  działa
an.error.occured <- TRUE
while(an.error.occured==TRUE){
  ok<-TRUE
  fit<-tryCatch(lmFit(v2,mmod1,method='robust', maxit=9999),
                warning=function(w){
                  
                  ok<<-FALSE
                })
  if (!ok)
  {
    nsv <-nsv-1
    svseq <- svaseq(to_sva,mod1,mod0,const=0,n.sv=nsv)
    mmod1 <- cbind(svseq$sv, mod1 )
    #colnames(mmod1) <- c(levels(factor(samples)), paste("X",1:nsv,sep=""))
    v2 <- voom(dge,mmod1,plot=FALSE)
    next}
  
  else { an.error.occured <- FALSE}
}   

newV=NULL
fsvaobj<-fsva(v$E,mod1,svseq,newdat=NULL) # jeszcze raz fsva na ostatecznej liczbie nsv
```

```{r, error=TRUE}
colnames(mmod1) <- c("X", "X", "X", "C7.H3", "C7.H5" ,"C7.N" ,"G.H3" ,"G.H5" ,"G.N" ,"TC.H3", "TC.H5" ,"TC.N")
design <- mmod1
```





Wizualna inspekcja
```{r, error=TRUE}

y<- dge
y <- estimateDisp(y,design, robust=TRUE)
plotBCV(y)
fit <- glmQLFit(y,design, robust=TRUE)
head(fit$coefficients)
plotQLDisp(fit)
summary(fit$df.prior)
```
#ANOVA LIKE TESTING
#https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf




Exploring differences between libraries
```{r, error=TRUE}

pch <- c(0,1,2,15,16,17, 18, 19 , 20)
colors <- rep(c("darkgreen", "red", "blue", "green", "orange"), 2)
plotMDS(y, col=colors[Group], pch=pch[Group])
legend("topright", legend=levels(Group), pch=pch, col=colors, ncol=2)

plotMDS(y, col=colors[Group], pch=pch[Group])

```




Utworzenie kontrastów:
```{r, error=TRUE, results='hide', message=FALSE}

conH3_H5 <- makeContrasts( (C7.H3+ G.H3 + TC.H3)/3 - (C7.H5+G.H5+TC.H5)/3, levels = design)
conH3_N <- makeContrasts( (C7.H3+ G.H3 + TC.H3)/3 - (C7.N+G.N+TC.N)/3, levels = design)
conH5_N <- makeContrasts( (C7.H5+ G.H5 + TC.H5)/3 - (C7.N+G.N+TC.N)/3, levels = design)

conC7_TC <- makeContrasts( (C7.H3+ C7.H5 + C7.N)/3 - (TC.H3+TC.H5+TC.N)/3, levels = design)
conC7_G <- makeContrasts( (C7.H3+ C7.H5 + C7.N)/3 - (G.H3+G.H5+G.N)/3, levels = design)
conTC_G <- makeContrasts( (TC.H3+ TC.H5 + TC.N)/3 - (G.H3+G.H5+G.N)/3, levels = design)

conC7 <- makeContrasts(
  C7.H3vsC7.H5 = C7.H3 - C7.H5,
  C7.NvsC7.H5 = C7.N - C7.H5,
  C7.NvsC7.H3 = C7.N - C7.H3, levels = design
  
)

conG  <- makeContrasts(
  G.H3vsG.H5 = G.H3 - G.H5,
  G.NvsG.H5 = G.N - G.H5,
  G.NvsG.H3 = G.N -G.H3, levels = design
  
)

conTC<- makeContrasts(
  TC.H3vsTC.H5 = TC.H3 - TC.H5,
  TC.NvsTC.H5 = TC.N - TC.H5,
  TC.NvsTC.H3 = TC.N -TC.H3, levels = design
  
)

my.contrasts <- makeContrasts(
  C7.H3vsC7.H5 = C7.H3 - C7.H5,
  C7.H3vsC7.N = C7.H3 - C7.N,
  C7.H5vsC7.N = C7.H5 - C7.N,
  G.H3vsG.H5 = G.H3 - G.H5, 
  G.H3vsG.N = G.H3 - G.N,
  G.H5vsG.N = G.H5 - G.N,
  TC.H3vsTC.H5 = TC.H3 - TC.H5,
  TC.H3vsTC.N = TC.H3 - TC.N, 
  TC.H5vsTC.N = TC.H5 - TC.N, levels = design 
)
contrasts_names <- c("C7.H3vsC7.H5" , "C7.H3vsC7.N", "C7.H5vsC7.N" , "G.H3vsG.H5" , "G.H3vsG.N" , "G.H5vsG.N",  "TC.H3vsTC.H5" , "TC.H3vsTC.N", "TC.H5vsTC.N"   )

```



```{r, error=TRUE}
qlf_test <- function(con, contr_name) {
  dir.create(paste0("results_",contr_name), showWarnings = FALSE)
  dir.create(paste0("results_",contr_name, "/qlf_test"), showWarnings = FALSE)
  qlf <- glmQLFTest(fit, contrast=con)
  res_qlf <- as.data.frame(qlf)[1:100,]
  write.csv(res_qlf, file = file.path(paste0("results_",contr_name, "/qlf_test"), "res_qlf.csv"))
  
  top_tags <- as.data.frame(topTags(qlf, n=100))
  write.csv(top_tags, file = file.path(paste0("results_",contr_name, "/qlf_test"), "qlf_topDEgenes.csv"))
  
  is.de <- decideTestsDGE(qlf)
  is.de05 <- decideTestsDGE(qlf, p.value=0.05)
  
  capture.output(summary(is.de), file = file.path(paste0("results_",contr_name, "/qlf_test"), "qlf_isde.txt"))
  capture.output(summary(is.de05), file = file.path(paste0("results_",contr_name, "/qlf_test"), "qlf_isde05.txt"))

  jpeg(file.path(paste0("results_",contr_name, "/qlf_test"), "MD_plot.jpg"))
  plotMD(qlf, status=is.de)
  dev.off()
  return(qlf)
}
```


```{r, error=TRUE}
LRT_test <- function(con, contr_name){
  dir.create(paste0("results_",contr_name), showWarnings = FALSE)
  dir.create(paste0("results_",contr_name, "/lrt_test"), showWarnings = FALSE)
  fit_LRT <- glmFit(y,design)
  lrt <- glmLRT(fit_LRT,contrast=con)
  res_lrt <- as.data.frame(lrt)[1:100,]
  write.csv(res_lrt, file = file.path(paste0("results_",contr_name, "/lrt_test"), "res_lrt.csv"))
  
  top_tags <- as.data.frame(topTags(lrt, n=100))
  write.csv(top_tags, file = file.path(paste0("results_",contr_name, "/lrt_test"), "lrt_topDEgenes.csv"))
  
  
  o <- order(lrt$table$PValue)
  capture.output(summary(de <- decideTestsDGE(lrt)), file = file.path(paste0("results_",contr_name, "/lrt_test"), "lrt_isde.txt"))
  print(summary(de <- decideTestsDGE(lrt)))
  detags <- rownames(y)[as.logical(de)]
  
  jpeg(file.path(paste0("results_",contr_name, "/lrt_test"), "lrt_plot.jpg"))
  plotSmear(lrt, de.tags=detags)
  abline(h=c(-1, 1), col="blue")
  dev.off()
  return(lrt)
}

```


```{r, error=TRUE} 
#Differential expression above a fold-change threshold
treat_test <- function(con, contr_name){
  dir.create(paste0("results_",contr_name), showWarnings = FALSE)
  dir.create(paste0("results_",contr_name, "/treat_test"), showWarnings = FALSE)
  tr <- glmTreat(fit, contrast=con, lfc=log2(1.5))
  res_tr <- as.data.frame(tr)[1:100,]
  write.csv(res_tr, file = file.path(paste0("results_",contr_name, "/treat_test"), "res_treat.csv"))
  
  print(topTags(tr))
  top_tags <- as.data.frame(topTags(tr, n=100))
  write.csv(top_tags, file = file.path(paste0("results_",contr_name, "/treat_test"), "treat_topDEgenes.csv"))
  
  
  is.de <- decideTestsDGE(tr)
  capture.output(summary(is.de), file = file.path(paste0("results_",contr_name, "/treat_test"), "treat_isde.txt"))
  print(summary(is.de))
  
  jpeg(file.path(paste0("results_",contr_name, "/treat_test"), "MD_plot.jpg"))
  plotMD(tr, status=is.de)
  dev.off()
  return(tr)
  }

```

```{r, error=TRUE}
#a heatmap to visualize the top 30 DE genes according to the test
heat_map <- function(contr_name, y, st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  logCPM <- cpm(y, prior.count=2, log=TRUE)
  rownames(logCPM) <- y$genes$SYMBOL
  colnames(logCPM) <- paste(y$samples$group, 1:3, sep="-")
  o <- order(st_test$table$PValue)
  logCPM <- logCPM[o[1:30],]
  
  jpeg(file.path(paste0("results_",contr_name,"/", folder_name), "heatmap.jpg"))
  coolmap(logCPM, margins=c(7,7), lhei=c(1,6), lwid=c(1,3))
  dev.off()
}
```

```{r, error=TRUE} 
go_function<- function(contr_name ,st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  go <- goana(st_test , geneid = "ENTREZID")
  
  #print(topGO(go, ont="BP", sort="Up", n=30))
  top_go <- topGO(go, sort="Up", n=30)
  top_go_df <- as.data.frame(top_go)
  write.csv(top_go_df, file = file.path(paste0("results_",contr_name, "/", folder_name), "go.csv"))
  
}
```


```{r, error=TRUE}
kegg_function<- function(contr_name ,st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  kegg <- kegga(st_test, geneid = "ENTREZID")
  top_kegg <- topKEGG(kegg)
  top_kegg_df <- as.data.frame(top_kegg)
  write.csv(top_kegg_df, file = file.path(paste0("results_",contr_name, "/", folder_name), "kegg.csv"))
}
```


```{r, error=TRUE}
heatmap_onlyDE <- function(top_tags, contr_name, folder_name){
  top_treat.topgenes <- top_tags$ENSEMBL[1:30]
  top_treat.topnames <- top_tags$SYMBOL[1:30]
  logCPM <- cpm(y, prior.count=2, log=TRUE)
  rownames(logCPM) <- v$genes$ENSEMBL
  
  nowy<-logCPM[as.vector(top_treat.topgenes),]
  rownames(nowy) <- top_treat.topnames
  jpeg(file.path(paste0("results_",contr_name, "/", folder_name), "heatmap_onlyDE.jpg"))
  coolmap(nowy ,margins=c(7,7), lhei=c(1,6), lwid=c(1,3),show.dendrogram = "both"  )#row both none column
  dev.off()
  }
```


```{r, error=TRUE}
heat_map_svobj <- function(top_tags, contr_name, folder_name){
  top_treat.topgenes <- top_tags$ENSEMBL[1:30]
  top_treat.topnames <- top_tags$SYMBOL[1:30]
  logCPM <- fsvaobj$db
  rownames(logCPM) <- v$genes$ENSEMBL
  
  nowy<-logCPM[as.vector(top_treat.topgenes),]
  rownames(nowy) <- top_treat.topnames
  jpeg(file.path(paste0("results_",contr_name, "/", folder_name), "heatmap_onlyDE_svobj.jpg"))
  coolmap(nowy ,margins=c(7,7), lhei=c(1,6), lwid=c(1,3),show.dendrogram = "both"  )#row both none column
  dev.off()

}


```


```{r, error=TRUE}
heat_map_svobj2 <- function(top_tags, contr_name, folder_name){
  top_treat.topgenes <- top_tags$ENSEMBL[1:100]
  top_treat.topnames <- top_tags$SYMBOL[1:30]
  logCPM <- fsvaobj$db
  rownames(logCPM) <- v$genes$ENSEMBL
  
  nowy<-logCPM[as.vector(top_treat.topgenes),]
  rownames(nowy) <- top_treat.topnames
  jpeg(file.path(paste0("results_",contr_name, "/", folder_name), "heatmap_onlyDE_svobj2.jpg"))
  coolmap(nowy ,margins=c(7,7), lhei=c(1,6), lwid=c(1,3),show.dendrogram = "both"  )#row both none column
  dev.off()
}

```


```{r, error=TRUE}
#a heatmap to visualize the top 30 DE genes according to the test
heat_map_svobj3<- function(contr_name, y, st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  logCPM <- fsvaobj$db
  rownames(logCPM) <- y$genes$SYMBOL
  colnames(logCPM) <- paste(y$samples$group, 1:3, sep="-")
  o <- order(st_test$table$PValue)
  logCPM <- logCPM[o[1:100],]
  
  #tiff(file.path(paste0("results_",contr_name,"/", folder_name), "heatmaponlyDE_svobj3.tiff"), units="in", width=10, height=10, res=300)
  jpeg((file.path(paste0("results_",contr_name,"/", folder_name), "heatmaponlyDE_svobj3.jpg")))
  coolmap(logCPM, margins=c(7,7), lhei=c(1,6), lwid=c(1,3))
  dev.off()
}
```


```{r, error=TRUE}
#a heatmap to visualize the top 30 DE genes according to the test
heat_map_svobj2<- function(contr_name, y, st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  logCPM <- fsvaobj$db
  rownames(logCPM) <- y$genes$SYMBOL
  colnames(logCPM) <- paste(y$samples$group, 1:3, sep="-")
  o <- order(st_test$table$PValue)
  logCPM <- logCPM[o[1:10],]
  
  jpeg(file.path(paste0("results_",contr_name,"/", folder_name), "heatmaponlyDE_svobj2.jpg"))
  coolmap(logCPM, margins=c(7,7), lhei=c(1,6), lwid=c(1,3))
  dev.off()
}
```

```{r, error=TRUE}
#a heatmap to visualize the top 30 DE genes according to the test
heat_map_svobj4<- function(contr_name, y, st_test, folder_name){
  dir.create(paste0("results_",contr_name, "/", folder_name), showWarnings = FALSE)
  logCPM <- fsvaobj$db
  rownames(logCPM) <- y$genes$SYMBOL
  colnames(logCPM) <- paste(y$samples$group, 1:3, sep="-")
  o <- order(st_test$table$PValue)
  logCPM <- logCPM[o[1:200],]
  
  jpeg(file.path(paste0("results_",contr_name,"/", folder_name), "heatmaponlyDE_svobj4.jpg"))
  coolmap(logCPM, margins=c(7,7), lhei=c(1,6), lwid=c(1,3))
  dev.off()
}
```

Funkcja do analizy DE
```{r, error=TRUE}
DE_analyse <- function(con, contr_name, y){
  
  qlf <- qlf_test(con, contr_name)
  try(lrt <- LRT_test(con, contr_name))
  tr <- treat_test(con, contr_name)

  heat_map(contr_name, y,  tr, "treat_test")
  heat_map(contr_name, y,  qlf, "qlf_test")
  heat_map(contr_name, y,  lrt, "lrt_test")

  go_function(contr_name, tr, "treat_test")
  go_function(contr_name, qlf, "qlf_test")
  go_function(contr_name, lrt, "lrt_test")

  kegg_function(contr_name, tr, "treat_test")
  kegg_function(contr_name, qlf, "qlf_test")
  kegg_function(contr_name, lrt, "lrt_test")
  
  top_tags_qlf <- as.data.frame(topTags(qlf, n=100))
  top_tags_treat <- as.data.frame(topTags(tr, n=100))
  try(top_tags_lrt <- as.data.frame(topTags(lrt, n=100)))
  

  heatmap_onlyDE(top_tags_qlf,contr_name, "qlf_test" )
  heatmap_onlyDE(top_tags_treat,contr_name, "treat_test" ) 
  try(heatmap_onlyDE(top_tags_lrt,contr_name, "lrt_test" ) )
  
  
  heat_map_svobj3(contr_name, y,  tr, "treat_test" )
  heat_map_svobj3(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj3(contr_name, y,  lrt, "lrt_test" )


  heat_map_svobj2(contr_name, y,  tr, "treat_test" )
  heat_map_svobj2(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj2(contr_name, y,  lrt, "lrt_test" )


  heat_map_svobj4(contr_name, y,  tr, "treat_test" )
  heat_map_svobj4(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj4(contr_name, y,  lrt, "lrt_test" )

}

```



```{r, error=TRUE}
DE_analyse_try <- function(con, contr_name, y){
  
  qlf <- qlf_test(con, contr_name)
  try(lrt <- LRT_test(con, contr_name))
  tr <- treat_test(con, contr_name)

  try(heat_map(contr_name, y,  tr, "treat_test"))
  try(heat_map(contr_name, y,  qlf, "qlf_test"))
  try(heat_map(contr_name, y,  lrt, "lrt_test"))

  try(go_function(contr_name, tr, "treat_test"))
  try(go_function(contr_name, qlf, "qlf_test"))
  try(go_function(contr_name, lrt, "lrt_test"))

  try(kegg_function(contr_name, tr, "treat_test"))
  try(kegg_function(contr_name, qlf, "qlf_test"))
  try(kegg_function(contr_name, lrt, "lrt_test"))
  
  top_tags_qlf <- as.data.frame(topTags(qlf, n=100))
  top_tags_treat <- as.data.frame(topTags(tr, n=100))
  try(top_tags_lrt <- as.data.frame(topTags(lrt, n=100)))
  

  heatmap_onlyDE(top_tags_qlf,contr_name, "qlf_test" )
  heatmap_onlyDE(top_tags_treat,contr_name, "treat_test" ) 
  try(heatmap_onlyDE(top_tags_lrt,contr_name, "lrt_test" ) )
  
  
  heat_map_svobj3(contr_name, y,  tr, "treat_test" )
  heat_map_svobj3(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj3(contr_name, y,  lrt, "lrt_test" )


  heat_map_svobj2(contr_name, y,  tr, "treat_test" )
  heat_map_svobj2(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj2(contr_name, y,  lrt, "lrt_test" )


  heat_map_svobj4(contr_name, y,  tr, "treat_test" )
  heat_map_svobj4(contr_name, y,  qlf, "qlf_test" )
  heat_map_svobj4(contr_name, y,  lrt, "lrt_test" )
}

```



Analiza anova - różnicujące geny ze względu na warunki dla konkretnych linii
```{r, error=TRUE}
DE_analyse_try(conC7, "DE_between_C7_line", y)
DE_analyse_try(conG, "DE_between_G_line", y)
DE_analyse_try(conTC, "DE_between_TC_line", y)
```


Analiza średnich:
```{r, error=TRUE}
DE_analyse(conH3_H5, "mean_H3vsH5", y)
DE_analyse(conH3_N, "mean_H3vsN", y)
DE_analyse(conH5_N, "mean_H5vsN", y)
DE_analyse(conC7_TC, "mean_C7vsTC", y)
DE_analyse(conC7_G, "mean_C7vsG", y)
DE_analyse(conTC_G, "mean_TCvsG", y) 


```



Pętla porównująca kontrasty DE
```{r, error=TRUE}
for (i in 1:length(contrasts_names)){
#for (i in 1:3){
  
  con <- my.contrasts[,contrasts_names[i]]
  DE_analyse(con, contrasts_names[i], y)
  
}

```
##############################################################################################################
Część do gene set tests -> zamienienie nazw genów na EntrezId oraz odfiltrowanie niedopasowanych nazw
```{r, error=TRUE}
y<- dge
rownames(y$counts) <- y$genes$ENTREZID
y <- y[!is.na(y$genes$ENTREZID), ]#odfiltrowanie brakujących entrezid

y <- estimateDisp(y,design, robust=TRUE)
plotBCV(y)
fit <- glmQLFit(y,design, robust=TRUE)
head(fit$coefficients)
plotQLDisp(fit)
summary(fit$df.prior)
rownames(fit$counts) <- fit$genes$ENTREZID

library(GO.db)
```


```{r}

gen_set_testing <- function(vector_of_go, contrast, con_name1, con_name2 ){
  #term <- select(GO.db, keys=vector_of_go, columns="TERM")
  #term
  Rkeys(org.Hs.egGO2ALLEGS) <- vector_of_go
  cyt.go.genes <- as.list(org.Hs.egGO2ALLEGS)
  print(cyt.go.genes)
  fry(y, index=cyt.go.genes, design=design, contrast=contrast)
  fit <- fit[!is.na(fit$genes$ENTREZID), ]
  res <- glmQLFTest(fit, contrast=contrast)
  index <- rownames(fit) %in% cyt.go.genes[[1]]
  barcodeplot(res$table$logFC, index=index, labels=c(con_name2,con_name1), main=vector_of_go[1])

  
}
```


```{r, error=TRUE}
cont_1 <- c("C7.H3", "C7.H3", "C7.H5","G.H3", "G.H3", "G.H5", "TC.H3", "TC.H3", "TC.H5")
cont_2 <- c("C7.H5", "C7.N", "C7.N","G.H5", "G.N", "G.N", "TC.H5", "TC.N", "TC.N")

cyt.go <- c("GO:0061718")

for (i in 1:length(contrasts_names)){

  cont <- my.contrasts[,contrasts_names[i]]
  gen_set_testing(cyt.go, cont, cont_1[i], cont_2[i] )

}

```

```{r, error=TRUE}
cyt.go <- c("GO:0005730")

for (i in 1:length(contrasts_names)){
  
  cont <- my.contrasts[,contrasts_names[i]]
  gen_set_testing(cyt.go, cont, cont_1[i], cont_2[i] )

}
```

```{r, error=TRUE}
cyt.go <- "GO:0046034"

for (i in 1:length(contrasts_names)){
  
  cont <- my.contrasts[,contrasts_names[i]]
  gen_set_testing(cyt.go, cont, cont_1[i], cont_2[i] )

}

```

```{r}
cyt.go <- c("GO:0005730", "GO:0061718")
contrast <- my.contrasts[,contrasts_names[2]]
# cont
# rownames(y$counts) <- y$genes$ENTREZID
#rownames(fit$counts) <- fit$genes$ENTREZID

#library(GO.db)

# term <- select(GO.db, keys=cyt.go, columns="TERM")
# term

# Rkeys(org.Hs.egGO2ALLEGS) <- cyt.go
# cyt.go.genes <- as.list(org.Hs.egGO2ALLEGS)
# cyt.go.genes

# cont <- my.contrasts[,contrasts_names[2]]
# cont
# fry(y, index=cyt.go.genes, design=design, contrast=cont)

fit <- fit[!is.na(fit$genes$ENTREZID), ]
res <- glmQLFTest(fit, contrast=cont)
index <- rownames(fit) %in% cyt.go.genes[[1]]
barcodeplot(res$table$logFC, index=index, labels=c("C7.N","C7.H3"), main=cyt.go[1])





cont <- my.contrasts[,contrasts_names[3]]
cont
fry(y, index=cyt.go.genes, design=design, contrast=cont)
res <- glmQLFTest(fit, contrast=cont)
index <- rownames(fit) %in% cyt.go.genes[[1]]
barcodeplot(res$table$logFC, index=index, labels=c("C7.N","C7.H5"), main=cyt.go[1])




```


Dlasza eksploracja wyników
```{r, error=TRUE}
cpm_log<- cpm(y, prior.count=2, log=TRUE)
cpm_log<- cpm(y, log = TRUE)
heatmap(cor(cpm_log))
#Another method to view the relationships between samples is principal components analysis (PCA).
pca <- prcomp(t(cpm_log), scale. = TRUE)
plot(pca$x[, 1], pca$x[, 2], pch = ".", xlab = "PC1", ylab = "PC2")
text(pca$x[, 1], pca$x[, 2], labels = colnames(cpm_log))
summary(pca)

#https://gist.github.com/jdblischak/11384914#file-rnaseq-de-tutorial-rmd
et <- exactTest(y)
results_edgeR <- topTags(et, n = nrow(y), sort.by = "none")
head(results_edgeR$table)
#How many genes are differentially expressed at an FDR of 10%?
sum(results_edgeR$table$FDR < .1)
plotSmear(et, de.tags = rownames(results_edgeR)[results_edgeR$table$FDR < .1])
abline(h = c(-2, 2), col = "blue")  


```
